[dynamo, logging] add default pt2 logging group

williamwen42 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #106417

Create a new logging group that enables "interesting" logging: graph breaks, recompiles, symbolic shapes, guards, source trace.

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @aakhundov

williamwen42 committed:
[dynamo, logging] add default pt2 logging group

[ghstack-poisoned]

anijain2305 commented on torch/_logging/_internal.py:
```diff
@@ -118,6 +118,18 @@ def clear(self):
+    "default": {
+        "graph_breaks": True,
+        "recompiles": True,
+        "dynamic": logging.DEBUG,
```
should be INFO.

anijain2305 reviewed (commented):
@ezyang Is this a good starting point?

ezyang commented:
Hmm, this would not have helped with the most recent SEV S355730 because the problem was not that folks hadn't enabled enough logs (they had; even dynamo was on debug), the problem was we literally were not generating any logs at all.

A direct fix for the most recent SEV would be to solve for these problems:

1. graph_break log doesn't always trigger on all graph breaks. We have logic in break_graph_if_unsupported but not for when a step fails
2.  Suppress errors / WON'T CONVERT don't show up as graph breaks, but people don't really care: anything that causes too many compiled graphs / prevents compilation is stuff they want to know about. So really these should also be reported as graph break as well

anijain2305 commented:
@ezyang This is not specifically for the Sev (though we need to solve those problems as well). This is about the people usually forget what to log. So, we can define one log artifact, and carefully print what to put in it, and then advertise that log artifact.

williamwen42 committed:
Update on "[dynamo, logging] add default pt2 logging group"


Create a new logging group that enables "interesting" logging: graph breaks, recompiles, symbolic shapes, guards, source trace.

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy chenyang78 aakhundov

[ghstack-poisoned]

mlazos commented:
> Hmm, this would not have helped with the most recent SEV S355730 because the problem was not that folks hadn't enabled enough logs (they had; even dynamo was on debug), the problem was we literally were not generating any logs at all.
> 
> A direct fix for the most recent SEV would be to solve for these problems:
> 
> 1. graph_break log doesn't always trigger on all graph breaks. We have logic in break_graph_if_unsupported but not for when a step fails
> 2. Suppress errors / WON'T CONVERT don't show up as graph breaks, but people don't really care: anything that causes too many compiled graphs / prevents compilation is stuff they want to know about. So really these should also be reported as graph break as well

WON'T CONVERT should show up no matter the log setting, (it is logged as an error I believe, if it isn't, it should be). We should not need to add new settings for that

ezyang commented:
@mlazos The problem is not that WON'T CONVERT is not logged (it is), but that it doesn't contain the string "Graph break" or "graph_break", and thus when people search logs they don't know that they need to search for "WON'T CONVERT" instead.

mlazos commented on torch/_logging/_internal.py:
```diff
@@ -118,6 +118,18 @@ def clear(self):
 log_registry = LogRegistry()
 log_state = LogState()
 
+# Groups of logging configurations that can be used with `set_logs`.
```
Can you put this in registrations and make a small API for it?
ie: `register_log_group(<alias>, <set of aliases to turn on>)`

I think this could be useful for people to create their own logging groups in the future

williamwen42 committed:
Update on "[dynamo, logging] add default pt2 logging group"


Create a new logging group that enables "interesting" logging: graph breaks, recompiles, symbolic shapes, guards, source trace.

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy chenyang78 aakhundov

[ghstack-poisoned]

ezyang commented on torch/_logging/_registrations.py:
```diff
@@ -25,3 +27,14 @@
+    {
+        "graph_breaks": True,
+        "recompiles": True,
+        "dynamic": logging.INFO,
```
So what happens if I add a conflicting log level for dynamic in, e.g., my TORCH_LOGS? What takes precedence?

mlazos replied:
Yeah I thought about this above, I think it would be reasonable to take the min of the conflict? that said I think your proposal below is a lot simpler.

ezyang commented:
I think I'm generally 👎  on logging groups as implemented here. If you want a suggested list of logs to enable, just make a single constant and rewrite fbcode sites to call `set_logging(torch._logging.DEFAULT_LOGGING)`. As is, this is a very invasive addition to the logging code with... no clear payoff, and many potential failure modes. Simple is better, especially for things like logs which are very awkward to unit test.

williamwen42 committed:
Update on "[dynamo, logging] add default pt2 logging group"


Create a new logging group that enables "interesting" logging: graph breaks, recompiles, symbolic shapes, guards, source trace.

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy chenyang78 aakhundov

[ghstack-poisoned]

williamwen42 commented:
@pytorchbot merge
